### ISSUE TITLE
Rescue Fedi - additionally implement domain blocks for users

### ISSUE THREAD

**Issue by AlfredSK** [friendica/friendica#7472]
### Expected behavior
Users should be able to block whole domains for themselves.

### Actual behavior
Every now and then there is a discussion about a bad node that exists somewhere in the Fediverse. Users feel the need to express that servers should block that domain. This is producing some pressure to act on the node's admin because the admin is the only person who can activate such a domain block.

What if we would have the possibility to add a domain block in the user settings? So users could decide for themselves to block all communication with a specific server/domain. Of cause, this should only block the communication for the one user. When the block is enabled the user should not get any content from that domain/server (including no contacts requests).

Site-wide domain blocks should - of cause - stay as they are.

### Friendica version you encountered the problem
2019.09-dev

### Friendica source (git, zip)
git

### PHP version
7.2

### SQL version
10.1 MariaDB

**Comment by MrPetovan** [comment]
of cause -> of course.

This repository is turning into an English class and I'm here for it.

---

This feature request has been discussed already and has been accepted on principles, the remaining tricky part is the actual implementation. The current URL check for blocks (node-wide domain/user) doesn't have a local user context because it didn't need it.

We already handle the contact block part for individual users, it would need to be extended to add a domain block feature as as well.

And we need a new page for that in user settings.

**Comment by AlfredSK** [comment]
> This feature request has been discussed already and has been accepted on principles

Yeah, I know. I was part of the discussion. But there was no issue for it yet.

> And we need a new page for that in user settings.

'Security and Privacy'? But we're also lacking a page where the user blocks are listed (when a user blocks someone w/o putting it in the contacts first). Maybe that would be also a good place for domain blocks.

**Comment by AlfredSK** [comment]
Link to discussion: https://forum.friendi.ca/display/0b6b25a8-115d-39f9-8d36-b88591040314

**Comment by MrPetovan** [comment]
Thanks for the link, it is the discussion I was referring to.

**Comment by AlfredSK** [comment]
...also we should make sure that a user cannot block the local node's domain. :-D

**Comment by MrPetovan** [comment]
I didn't consider this, thanks for this valuable input.
